---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  fig.width = 10,
  fig.height = 6
)
```

# tcgaViz

<!-- badges: start -->
[![CRAN status](https://www.r-pkg.org/badges/version/tcgaViz)](https://CRAN.R-project.org/package=tcgaViz)
[![R build status](https://github.com/ecamenen/tcgaViz/workflows/R-CMD-check/badge.svg)](https://github.com/ecamenen/tcgaViz/actions)
[![Codecov test coverage](https://codecov.io/gh/ecamenen/tcgaViz/branch/develop/graph/badge.svg)](https://codecov.io/gh/ecamenen/tcgaViz?branch=master)
<!-- badges: end -->

Differential analysis of tumor tissue immune cell type abundance based on
RNASeq gene-level expression from The Cancer Genome Atlas (TCGA) database.

## Installation

Required:
- Softwares : R (â‰¥ 3.3.0)
- R libraries : see the [DESCRIPTION](https://github.com/ecamenen/tcgaViz/blob/develop/DESCRIPTION) file.

You can install the development version from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("ecamenen/tcgaViz")
```

## Launch the Shiny server
Open R, then run:
```r
runApp()
```

## Docker

###### Pull
```
docker pull eucee/tcga-viz
```

###### (Local installation)
```
docker build -t tcga-viz .
```

###### Run in command-line
```
docker run --rm \
  -p 127.0.0.1:8787:8787 \
  -v $(pwd):/home/rstudio \
  -e ROOT=TRUE \
  -e DISABLE_AUTH=true \
  tcga-viz
```

## Example

###### Load the dataset
(with the same cancer type and TCGA sample identifiers)
```{r setup}
library(tcgaViz)
library(ggpubr, quietly = TRUE)
data(tcga)
head(tcga$genes)
head(tcga$cells$Cibersort_ABS)
```

###### Violin plot of cell subtypes
(and significance of a Wilcoxon adjusted test according to the expression
 level [high or low] of a selected gene)
```{r plot}
(df <- convert2biodata(
  algorithm = "Cibersort_ABS",
  disease = "breast invasive carcinoma",
  tissue = "Primary Tumor",
  gene_x = "ICOS"
))
(stats <- calculate_pvalue(df))
plot(df, stats = stats)
```

###### Advanced parameters
```{r advanced}
(df <- convert2biodata(
  algorithm = "Cibersort_ABS",
  disease = "breast invasive carcinoma",
  tissue = "Primary Tumor",
  gene_x = "ICOS",
  stat = "quantile"
))
(stats <- calculate_pvalue(
    df,
    method_test = "t_test",
    method_adjust = "bonferroni",
    p_threshold = 0.01
))
plot(
    df,
    stats = stats,
    type = "boxplot",
    dots = TRUE,
    xlab = "Expression level of the 'ICOS' gene by cell type",
    ylab = "Percent of relative abundance\n(from the Cibersort_ABS algorithm)",
    title = toupper("Differential analysis of immune cell type abundance
    based on RNASeq gene-level expression from The Cancer Genome Atlas"),
    axis.text.y = element_text(size = 8, hjust = 0.5),
    plot.title =  element_text(face = "bold", hjust = 0.5),
    plot.subtitle =  element_text(size = , face = "italic", hjust = 0.5),
    draw = FALSE
) + labs(
    subtitle = paste("Breast Invasive Carcinoma (BRCA; Primary Tumor):",
    "Student's t-test with Bonferroni (P < 0.01)")
)
```